lib/ansible/modules/network/iosxr/iosxr_logging.py

```diff
@@ -361,8 +361,10 @@ def main():
     result['warnings'] = warnings
 
     if commands:
-        if not module.check_mode:
-            load_config(module, commands, result['warnings'], commit=True)
+        commit = not module.check_mode
+        diff = load_config(module, commands, commit=commit)
+        if diff:
+            result['diff'] = dict(prepared=diff)
```

Comment:
Diff should be added to result if --diff option is enabled ie. `module._diff` right?

Reply:
That is handled in the `load_config` method in utils